scripts/exgdas_enkf_sfc.sh

```diff
@@ -193,37 +229,58 @@ if [ $DOSFCANL_ENKF = "YES" ]; then
-                COM_ATMOS_RESTART_MEM:COM_ATMOS_RESTART_TMPL
+            RUN="${GDUMP_ENS}" MEMDIR=${gmemchar} YMD=${PDY} HH=${cyc} declare_from_tmpl \
+                COMIN_SNOW_ANALYSIS_MEM:COM_SNOW_ANALYSIS_TMPL
 
             RUN="${GDUMP_ENS}" MEMDIR=${gmemchar} YMD=${gPDY} HH=${gcyc} declare_from_tmpl \
-                COM_ATMOS_RESTART_MEM_PREV:COM_ATMOS_RESTART_TMPL
+                COMIN_ATMOS_RESTART_MEM_PREV:COM_ATMOS_RESTART_TMPL
 
-            [[ ${TILE_NUM} -eq 1 ]] && mkdir -p "${COM_ATMOS_RESTART_MEM}"
+            # determine where the input snow restart files come from
+            if [[ "${DO_JEDISNOWDA:-}" == "YES" ]]; then
+                sfcdata_dir="${COMIN_SNOW_ANALYSIS_MEM}"
+            else
+                sfcdata_dir="${COMIN_ATMOS_RESTART_MEM_PREV}"
+            fi
 
-            ${NCP} "${COM_ATMOS_RESTART_MEM_PREV}/${PDY}.${cyc}0000.sfc_data.tile${n}.nc" \
-                "${COM_ATMOS_RESTART_MEM}/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc"
-            ${NLN} "${COM_ATMOS_RESTART_MEM_PREV}/${PDY}.${cyc}0000.sfc_data.tile${n}.nc" \
+            ${NCP} "${sfcdata_dir}/${PDY}.${cyc}0000.sfc_data.tile${n}.nc" \
                 "${DATA}/fnbgsi.${cmem}"
-            ${NLN} "${COM_ATMOS_RESTART_MEM}/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc" \
-                "${DATA}/fnbgso.${cmem}"
-            ${NLN} "${FIXorog}/${CASE}/${CASE}_grid.tile${n}.nc"      "${DATA}/fngrid.${cmem}"
-            ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc" "${DATA}/fnorog.${cmem}"
+            ${NCP} "${DATA}/fnbgsi.${cmem}" "${DATA}/fnbgso.${cmem}"
+            ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${n}.nc"      "${DATA}/fngrid.${cmem}"
+            ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc" "${DATA}/fnorog.${cmem}"
 
         done
 
         CDATE="${PDY}${cyc}" ${CYCLESH}
         export err=$?; err_chk
 
+        # Copy outputs from DATA to COMOUT
+        for imem in $(seq 1 "${NMEM_ENS}"); do
+            smem=$((imem + mem_offset))
+            if (( smem > NMEM_ENS_MAX )); then
+               smem=$((smem - NMEM_ENS_MAX))
+            fi
+            gmemchar="mem"$(printf %03i "${smem}")
+            cmem=$(printf %03i "${imem}")
+            memchar="mem${cmem}"
+
+            MEMDIR=${memchar} YMD=${PDY} HH=${cyc} declare_from_tmpl \
+                COM_ATMOS_RESTART_MEM:COM_ATMOS_RESTART_TMPL
+
+            [[ ${TILE_NUM} -eq 1 ]] && mkdir -p "${COM_ATMOS_RESTART_MEM}"
+
+            ${NCP} "${DATA}/fnbgso.${cmem}" "${COM_ATMOS_RESTART_MEM}/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc"
+
+        done
+
     done
 fi
 
 ################################################################################
 
 ################################################################################
 # Postprocessing
-cd $pwd
-[[ $mkdata = "YES" ]] && rm -rf $DATA
+cd "${pwd}" || exit 1
+[[ ${mkdata} = "YES" ]] && rm -rf "${DATA}"
```

Comment:
ex-scripts should not be deleting `$DATA`; the j-job already does that.

Reply:
Yeah I think this was already here, I'll remove but note it just in case it breaks something